File: app/src/main/java/com/techventus/wikipedianews/logging/Toaster.java
package com.techventus.wikipedianews.logging;

import android.content.Context;
import android.widget.Toast;
import com.techventus.wikipedianews.WikiApplication;


/**
 * Created by josephmalone on 15-09-24.
 */
public class Toaster
{

	public static final int LENGTH_LONG = Toast.LENGTH_LONG;
	public static final int LENGTH_SHORT = Toast.LENGTH_SHORT;

	// Prefix to help filtering of logcat for output from our app
	private static final String PREFIX = "#DEBUG# ";
	// ***Don't change this*** it's now set on app startup based on the debuggable flag in the manifest
	private static boolean showToast;

	/**
	 * Return whether logging is enabled (implies the debuggable flag is set)
	 *
	 * @return true if logging is enabled
	 */
	public static boolean getShowlog()
	{
		return showToast;
	}

	public static void setEnabled(Boolean enabled)
	{
		showToast = enabled;
		Logger.i("Logger", "Toaster ENABLED.");
	}


	public static void show(int stringId)
	{
		if (showToast)
		{
			Toast.makeText(WikiApplication.getInstance(), stringId, Toast.LENGTH_LONG).show();
		}
	}

	public static void show(Context context, int StringId, int lengthInMillis)
	{
		if (showToast && context != null)
		{
			Toast.makeText(context, StringId, lengthInMillis).show();
		}
	}

	public static void show(Context context, String string, int lengthInMillis)
	{
		if (showToast && context != null)
		{
			Toast.makeText(context, string, lengthInMillis).show();
		}
	}
}

File: app/src/main/java/com/techventus/wikipedianews/logging/Toaster.kt
package com.techventus.wikipedianews.logging

import android.content.Context
import android.widget.Toast
import com.techventus.wikipedianews.App

/**
 * Created by josephmalone on 15-09-24.
 */
object Toaster {
    const val LENGTH_LONG = Toast.LENGTH_LONG
    const val LENGTH_SHORT = Toast.LENGTH_SHORT

    // Prefix to help filtering of logcat for output from our app
    private const val PREFIX = "#DEBUG# "

    // ***Don't change this*** it's now set on app startup based on the debuggable flag in the manifest
    @JvmField
    var showToast = false

    /**
     * Return whether logging is enabled (implies the debuggable flag is set)
     *
     * @return true if logging is enabled
     */
    @JvmStatic
    fun getShowlog(): Boolean {
        return showToast
    }

    @JvmStatic
    fun setEnabled(enabled: Boolean) {
        showToast = enabled
        Logger.i("Logger", "Toaster ENABLED.")
    }

    @JvmStatic
    fun show(stringId: Int) {
        if (showToast) {
            App.getInstance()?.let {
                Toast.makeText(it, stringId, Toast.LENGTH_LONG).show()
            }
        }
    }

    @JvmStatic
    fun show(context: Context?, stringId: Int, lengthInMillis: Int) {
        if (showToast && context != null) {
            Toast.makeText(context, stringId, lengthInMillis).show()
        }
    }

    @JvmStatic
    fun show(context: Context?, string: String?, lengthInMillis: Int) {
        if (showToast && context != null && string != null) {
            Toast.makeText(context, string, lengthInMillis).show()
        }
    }
}
